### ISSUE TITLE
Password forgotten

### ISSUE THREAD

**Issue by kreativmind** [xenocrat/chyrp-lite#49]
I am trying to remember my password but I can't. I am also trying to figure out which type of encryption are you using so I can put in a password again for my admin account so I can log back in. Thanks.

**Comment by xenocrat** [comment]
Chyrp Lite uses PHP's [`crypt()`](http://php.net/manual/en/function.crypt.php) function with a SHA-512 hash.

Here's a script you can use to generate a new password hash. Save this as e.g. "new_pw.php" in your Chyrp Lite install directory and visit it in your web browser. It will generate and display a hash for "my new password".

    <?php
        require_once "includes".DIRECTORY_SEPARATOR."common.php";
    
        $salt = random(16);
        $prefix = '$6$'; # Prefix for SHA-512.
        echo crypt("my new password", $prefix.$salt);

You can take that hash and paste it into your password value field in the users database table.

**Comment by kreativmind** [comment]
Thanks bro! Had to adjust the GroupID and Approved fields also before it would successfully let me get back into admin. Working now.

**Comment by dajare** [comment]
Is Chyrp Lite's login case-sensitive? I am positive I installed with an account of "admin" (lower case) but could not log in to the site, because of bad password. Neither could I generate an email to recover password/account. So I found this issue, and checked my db (SQLite). I found the user name as "Admin": I lower-cased it, and my password worked.

**IF** CL is case-sensitive for user names, then it really should't be upper-casing the name! I took a quick look at the `ucfirst()` occurrences in the code, and couldn't spot a culprit ... and it *could* be user error 😬 ! But I've installed dozens of CMS's, and would be surprised if I typed "Admin" rather than "admin" — I have never of my own volition used the former! 

Anyway, I thought it worth mentioning, and good to have this issue as part of Chyrp Lite's "knowledgebase".
